cpu/stm32l1/include/periph_cpu.h

```diff
@@ -108,6 +108,19 @@ typedef struct {
 #define PERIPH_SPI_NEEDS_TRANSFER_REGS
 /** @} */
 
+/**
+ * @brief   I2C device configuration
+ */
+typedef struct {
+    I2C_TypeDef *dev;       /**< pointer to the used I2C device */
+    gpio_t scl_pin;         /**< pin used for SCL */
+    gpio_t sda_pin;         /**< pin used for SDA */
+    uint8_t af_scl_sda;     /**< SCL or SDA AF */
+    uint8_t ev_irqn;        /**< I2C device Event Interrupt */
+    uint8_t er_irqn;        /**< I2C device Error Interrupt */
+    uint32_t clk_en;        /**< Clock enable register */
```

Comment:
@ReneHerthel you could add:
``` C
mutex_t mtx; /**< i2c config mutex */
```
here, and change the initialization of the config array [1] by appending the static mutex initializer `MUTEX_INIT`, e.g.:
``` C
{I2C2, GPIO_PIN(PORT_B, 10), GPIO_PIN(PORT_B, 11), 4,
     I2C2_EV_IRQn, I2C2_ER_IRQn, RCC_APB1ENR_I2C2EN, MUTEX_INIT},
```
(though this is untested, so don't know if this works)

[1] https://github.com/RIOT-OS/RIOT/pull/4096/files#diff-b7ca3df665fccc20bb1375fdb885cafbR246